monads/test/Spec.hs

```diff
@@ -1,2 +1,76 @@
+import Test.Tasty
+import Test.Tasty.HUnit
+import qualified Data.Map as M
+import State.MyState
+import HW.Compiler
+import HW.Eval
+import HW.StackMachine 
+
+pushFive :: StackInstr String
+pushFive = PushNum 5
+
+addInstr :: StackInstr String
+addInstr = Add
+
```

Comment:
There is a lot of code duplication in this file. Could you try to get rid of it?